components/proxy/src/main/java/com/hotels/styx/startup/StyxServerComponents.java

```diff
@@ -262,7 +278,14 @@ public StyxServerComponents build() {
     public interface LoggingSetUp {
         LoggingSetUp DO_NOT_MODIFY = environment -> {
         };
-        LoggingSetUp FROM_CONFIG = environment -> setUpLogging(environment.configuration().logConfigLocation());
+        LoggingSetUp FROM_CONFIG = environment -> setUpLogging(
+                Paths.get(environment
+                        .startupConfig()
+                        .logConfigLocation()
+                        .url()
+                        .getFile())
+                        .toString()
+        );
```

Comment:
I believe this is the only reason to add the `startupConfig` into `Environment`. We should look into moving the `FROM_CONFIG` out of this file. Perhaps it could be just a lambda in the `StyxServer` class.

If so, then `startupConfig` could be removed from environment. This would be a big plus as it would narrow down `startupConfig` usage to just admin interface where it is used merely to display the config file locations.

Reply:
Approving regardless of the above comment.